tensorflow/compiler/mlir/xla/transforms/map_lhlo_to_scalar_op.h

```diff
@@ -148,44 +155,173 @@ inline Optional<CmpIPredicate> getIntCmpPredicate(
+    FloatType src = sourceType.cast<FloatType>();
+    FloatType res = targetType.cast<FloatType>();
+    if (src.getWidth() > res.getWidth()) {
+      op = b.create<mlir::FPTruncOp>(lhlo_op.getLoc(), result_types, block_args,
+                                     mlir::None);
+    } else if (src.getWidth() < res.getWidth()) {
+      op = b.create<mlir::FPExtOp>(lhlo_op.getLoc(), result_types, block_args,
+                                   mlir::None);
+    } else {
+      return block_args.front();
+    }
+  } else if (sourceType.isa<IntegerType>() && targetType.isa<IntegerType>()) {
+    IntegerType src = sourceType.cast<IntegerType>();
+    IntegerType res = targetType.cast<IntegerType>();
+    if (src.getWidth() > res.getWidth()) {
+      op = b.create<mlir::TruncateIOp>(lhlo_op.getLoc(), result_types,
+                                       block_args, mlir::None);
+    } else if (src.getWidth() < res.getWidth()) {
+      op = b.create<mlir::ZeroExtendIOp>(lhlo_op.getLoc(), result_types,
+                                         block_args, mlir::None);
+    } else {
+      return block_args.front();
+    }
+  }
+  // TODO: Add other primitive type conversions
+  // else if (mlir::FpToSiOp::areCastCompatible(sourceType, targetType)) {
+  //   return b.create<mlir::FpToSiOp>(lhlo_op.getLoc(), result_types,
+  //   block_args,
+  //                                   mlir::None);
+  // }
+
+  return (op) ? op->getResult(0) : nullptr;
+}
+
+template <>
+inline Value* MapLhloOpToStdScalarOp<xla_lhlo::ExpOp>(
     xla_lhlo::ExpOp lhlo_op, ArrayRef<Type> result_types,
     ArrayRef<Value*> block_args, OpBuilder b) {
   Type element_type = block_args.front()->getType();
-  return element_type.isa<FloatType>()
-             ? b.create<::mlir::ExpOp>(lhlo_op.getLoc(), result_types,
-                                       block_args, mlir::None)
-             : nullptr;
+  if (element_type.isa<FloatType>()) {
+    Operation* op = b.create<::mlir::ExpOp>(lhlo_op.getLoc(), result_types,
+                                            block_args, mlir::None);
+    return op->getResult(0);
+  }
+  return nullptr;
+}
+
+template <>
+inline Value* MapLhloOpToStdScalarOp<xla_lhlo::NegOp>(
+    xla_lhlo::NegOp lhlo_op, ArrayRef<Type> result_types,
+    ArrayRef<Value*> block_args, OpBuilder b) {
+  Type element_type = block_args.front()->getType();
+  if (element_type.isa<FloatType>()) {
+    Operation* op = b.create<::mlir::NegFOp>(lhlo_op.getLoc(), result_types,
+                                             block_args, mlir::None);
+    return op->getResult(0);
+  }
+  return nullptr;
+}
+
+template <>
+inline Value* MapLhloOpToStdScalarOp<xla_lhlo::SignOp>(
+    xla_lhlo::SignOp lhlo_op, ArrayRef<Type> result_types,
+    ArrayRef<Value*> block_args, OpBuilder b) {
+  Type element_type = block_args.front()->getType();
+  Value* one = b.create<mlir::ConstantFloatOp>(
+      lhlo_op.getLoc(), APFloat(1.0), FloatType::getF32(b.getContext()));
```

Comment:
You can use `b.getFloatAttribute(element_type, 1.0)` here. The construction of one should be moved into the conditional.